RLTest/loader.py

```diff
@@ -96,48 +101,53 @@ def load_spec(self, arg):
             sys.path.append(dirname)
 
         module_name, _ = os.path.splitext(os.path.basename(filename))
+        toplevel_filter = None
+        subfilter = None
         if varname:
             if '.' in varname:
-                self.toplevel_filter, self.subfilter = varname.split('.')
+                toplevel_filter, subfilter = varname.split('.')
             else:
-                self.toplevel_filter = varname
+                toplevel_filter = varname
 
-        self.load_files(dirname, module_name)
+        self.load_files(dirname, module_name, toplevel_filter, subfilter)
 
-    def load_files(self, module_dir, module_name):
+    def load_files(self, module_dir, module_name, toplevel_filter=None, subfilter=None):
         filename = '%s/%s.py' % (module_dir, module_name)
-        module_file = open(filename, 'r')
-        module = imp.load_module(module_name, module_file, filename,
-                                 ('.py', 'r', imp.PY_SOURCE))
-        for symbol in dir(module):
-            if not self.filter_modulevar(symbol):
-                continue
-
-            obj = getattr(module, symbol)
-            if inspect.isclass(obj):
-                methnames = [mname for mname in dir(obj)
-                             if self.filter_method(mname)]
-                self.tests.append(TestClass(filename, symbol, module_name, methnames))
-            elif inspect.isfunction(obj):
-                self.tests.append(TestFunction(filename, symbol, module_name))
+        try:
```

Comment:
The try/catch isn't granular enough

Reply:
This is problematic from another respect: we might create too much noise here. Maybe it's better to fail silently. Need to think.